src/components/views/right_panel/UserInfo.tsx

```diff
@@ -1573,7 +1565,7 @@ const UserInfoHeader: React.FC<{
     }
 
     let statusLabel = null;
-    if (statusMessage) {
+    if (SettingsStore.getValue("feature_custom_status") && statusMessage) {
```

Comment:
IMO the hook should watch `SettingsStore.getValue("feature_custom_status")` instead, handling even more de-duplication of code. Also means it can not set up event emitter handles when the feature is disabled (if you pass a falsey eventEmitter as the first arg into a `useEventEmitter*` func it will no-op)

Let me know if that needs a better explanation

Reply:
Would something like this be sufficient?

```ts
export const useUserStatusMessage = (user?: Member): string => {
    const cli = useContext(MatrixClientContext);
    const enabled = useFeatureEnabled("feature_custom_status");
    return useEventEmitterState(enabled && getUser(cli, user), "User.unstable_statusMessage", () => {
        return getStatusMessage(cli, user);
    });
};

```

Reply:
Yup, thats what I had in mind, though would be worth a test :P

Reply:
Hmm, it seems to work though it doesn't hide the statuses from the right panel when I disable the feature, which I assumed it would... Should it do that?